src/Microsoft.Management.Configuration.Processor/ProcessorEnvironments/HostedEnvironment.cs

```diff
@@ -256,41 +256,54 @@ public void ImportModule(string path)
                 { Parameters.Name, unitInternal.Unit.UnitName },
             };
 
-            if (unitInternal.Module is not null)
+            // Don't use ModuleSpecification here. Each parameter is independent and
+            // we need version even if a module was not specified.
+            string? moduleName = unitInternal.GetDirective<string>(DirectiveConstants.Module);
+            string? semanticVersion = unitInternal.GetDirective<string>(DirectiveConstants.Version);
+            string? semanticMinVersion = unitInternal.GetDirective<string>(DirectiveConstants.MinVersion);
+            string? semanticMaxVersion = unitInternal.GetDirective<string>(DirectiveConstants.MaxVersion);
+            string? repository = unitInternal.GetDirective<string>(DirectiveConstants.Repository);
+            bool allowPrerelease = unitInternal.GetDirective(DirectiveConstants.AllowPrerelease);
+
+            if (!string.IsNullOrEmpty(moduleName))
             {
-                parameters.Add(Parameters.ModuleName, unitInternal.Module.Name);
+                parameters.Add(Parameters.ModuleName, moduleName);
+            }
 
-                if (unitInternal.Module.Version is not null)
-                {
-                    parameters.Add(Parameters.MinimumVersion, unitInternal.Module.Version);
-                }
+            if (!string.IsNullOrEmpty(semanticVersion))
+            {
+                parameters.Add(Parameters.RequiredVersion, semanticVersion);
+            }
 
-                if (unitInternal.Module.MaximumVersion is not null)
-                {
-                    parameters.Add(Parameters.MaximumVersion, unitInternal.Module.MaximumVersion);
-                }
+            if (!string.IsNullOrEmpty(semanticMinVersion))
+            {
+                parameters.Add(Parameters.MinimumVersion, semanticMinVersion);
+            }
 
-                if (unitInternal.Module.RequiredVersion is not null)
-                {
-                    parameters.Add(Parameters.RequiredVersion, unitInternal.Module.RequiredVersion);
-                }
+            if (!string.IsNullOrEmpty(semanticMaxVersion))
+            {
+                parameters.Add(Parameters.MaximumVersion, semanticMaxVersion);
             }
 
-            string? repository = unitInternal.GetDirective(DirectiveConstants.Repository);
             if (!string.IsNullOrEmpty(repository))
             {
                 parameters.Add(Parameters.Repository, repository);
             }
 
             using PowerShell pwsh = PowerShell.Create(this.Runspace);
 
-            // TODO: Implement prerelease directive.
+            pwsh.AddCommand(Commands.FindDscResource)
+                .AddParameters(parameters);
+
+            if (allowPrerelease)

```

Comment:
Why isn't this placed into the `parameters` like the others?

Reply:
I was to lazy to see if a switch with bool was supported in the parameters hashtable